src/org/sosy_lab/java_smt/SolverContextFactory.java

```diff
@@ -366,4 +366,42 @@ public static SolverContext createSolverContext(Solvers solver)
             NativeLibraries::loadLibrary)
         .generateContext(solver);
   }
+
+  /**
+   * Create a new optimization prover environment.
+   * @param context The solver context
+   * @param options The prover options
+   * @return An optimization prover environment
+   * @throws SolverOptimizationException if the solver doesn't support optimization
+   */
+  public static OptimizationProverEnvironment createOptimizationProver(
```

Comment:
The `SolverContextFactory` is intended to construct a `SolverContext`. This method violates this constraint, as it constructs a `ProverEnvironment`.